pkg/agent/controller/bgp/controller.go

```diff
@@ -397,13 +431,14 @@ func (c *Controller) syncBGPPolicy(ctx context.Context) error {
 
 		// Initialize the BGPPolicy state to store the new BGP server, BGP policy name, listen port, local ASN, and router ID.
 		c.bgpPolicyState = &bgpPolicyState{
-			bgpServer:     bgpServer,
-			bgpPolicyName: bgpPolicyName,
-			routerID:      routerID,
-			listenPort:    listenPort,
-			localASN:      localASN,
-			routes:        make(map[bgp.Route]RouteMetadata),
-			peerConfigs:   make(map[string]bgp.PeerConfig),
+			bgpServer:           bgpServer,
```

Comment:
Not directly with this PR, but I noticed that `effectivePolicy.Spec.BGPPeers` is not a required field, is this expected? If yes, it looks like there are some improvements needed in `reconcileBGPPeers` to make sure `nil` is checked.

Reply:
No peer is allowed in BGP; nil is considered in `reconcileBGPPeers`.  See `getPeerConfigs`.